store/src/java/com/zimbra/cs/service/mail/SendMsg.java

```diff
@@ -282,6 +292,48 @@ public Element handle(Element request, Map<String, Object> context, QName respQn
            }
     }
 
+    /**
+     * ZBUG-2310
+     * When we send mail first time to a new email id, it new contact in 
+     * AUTO_CONTACTS folder, and than we modify contact display name, 
+     * then this method is updating the contact record, so that we can see 
+     * updated display name in emailed contacts after sending email.
+     * @param acct This is the Account object to initialise object
+     * @param zsc  This is the ZimbraSoapContext object to initialise object
+     * @param octxt This is the OperationContext object to initialise object
+     * @param mm This is the MimeMessage object to get Recipient list
+     * @param mbox This is the Mailbox object to call modify contact
+     * @return nothing
+     */
+    public void modifyContactName(Account acct, ZimbraSoapContext zsc, OperationContext octxt, MimeMessage mm, Mailbox mbox)
```

Comment:
any specific reason that we have added code to modify contact here but code for creating contact is in some different file https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/mailbox/MailSender.java#L627

Reply:
Yes, actually we were in need to get contact details especially contact Id from DB to modify, for which I used ContactAutoComplete() constructor, which requires Account acct, ZimbraSoapContext zsc, OperationContext octxt objects to initialise it, here in this file we are able to get all these values. So once after email is sent then, wrote the code to modify it.